libraries/AP_DDS/AP_DDS_Client.h

```diff
@@ -0,0 +1,93 @@
+#pragma once
+
+#if AP_DDS_ENABLED
+
+#include "uxr/client/client.h"
+#include "ucdr/microcdr.h"
+#include "generated/Time.h"
+#include "AP_DDS_Generic_Fn_T.h"
+
+#include <AP_HAL/AP_HAL.h>
+#include <AP_HAL/Scheduler.h>
+#include <AP_HAL/Semaphores.h>
+#include <AP_AHRS/AP_AHRS.h>
+
+#include "fcntl.h"
+
+#include <AP_Param/AP_Param.h>
+#define STREAM_HISTORY 8
+#define BUFFER_SIZE_SERIAL UXR_CONFIG_SERIAL_TRANSPORT_MTU * STREAM_HISTORY
+
+extern const AP_HAL::HAL& hal;
+
+class AP_DDS_Client
+{
+
+private:
+
+    // Serial Allocation
+    uxrSerialTransport serial_transport; // client uxr serial transport
+    uxrSession session; //Session
+
+    // Input Stream
+    uint8_t input_reliable_stream[BUFFER_SIZE_SERIAL];
+    uxrStreamId reliable_in;
+
+    // Output Stream
+    uint8_t output_reliable_stream[BUFFER_SIZE_SERIAL];
+    uxrStreamId reliable_out;
+
+    // Topic
+    builtin_interfaces_msg_Time time_topic;
+
+    // Data Writer
+    const uxrObjectId dwriter_id = {
+        .id = 0x01,
+        .type = UXR_DATAWRITER_ID
+    };
+
+    HAL_Semaphore csem;
+
+    // connection parametrics
+    bool connected = true;
+
+    AP_Int8 xrce_type;
+public:
+    // Constructor
+    AP_DDS_Client();
+
+    void main_loop(void);
+
+    //
+    [[nodiscard]] bool init();
```

Comment:
what are all the nodiscards here for (and in AP_Vehicle). Why would we want to keep unused symbols?

Reply:
It is used; it's enforced by the compiler to be used. Throwing away the result of init() and ignoring it would be bad.

Reply:
ahh, for some reason I thought this was a linker directive, looks like it is equivalent to WARN_IF_UNUSED in ArduPilot, which is __attribute__ ((warn_unused_result)) for GNUC.
I wonder if [[nodiscard]] works with all our compilers?

Reply:
If CI passes. What's the behavior if it's not implemented? It's ignored? 

IF so, that's fine. Because the other CI jobs will catch the unused return value.

Reply:
Resolution is to use the macro already in the build system.

Reply:
Used the macro in a52db916f7f758c580cc10a69315f00273a0e960.